infrastructure/infrastructure/src/main/java/alabno/msfeedback/haskellupdater/HaskellMarkerRunner.java

```diff
@@ -0,0 +1,34 @@
+package alabno.msfeedback.haskellupdater;
+
+import alabno.utils.NumericUtils;
+
+/**
+ * Refreshes the training suite at period intervals
+ *
+ */
+public class HaskellMarkerRunner implements Runnable {
+
+    private static final int INTERVAL_MINUTES = 5;
+    
+    private final HaskellMarkerUpdater updater;
+
+    // package visibility
+    HaskellMarkerRunner(HaskellMarkerUpdater updater) {
+        this.updater = updater;
+    }
+    
+    @Override
+    public void run() {
+        System.out.println("HaskellMarkerRunner: starting...");
+        try {
+            while (true) {
+                updater.updateTraining();
+                Thread.sleep(NumericUtils.minutesToMillis(INTERVAL_MINUTES));
```

Comment:
if i get this right, are you continuously updating the training every 5 minutes? why so often? and... that while true probably looks worse than the reflection used in comp anal